### PR TITLE
Add spacemacs emoji automatic reaction

### PR DESCRIPTION
**Pull request by theckman**
This adds the new `spacemacs` emoji reaction to messages with `spacemacs` in it.

Signed-off-by: Tim Heckman <t@heckman.io>

### CODE REVIEW

File: bot/bot.go [hunk 1]
```diff
@@ -212,19 +213,25 @@ func (b *Bot) specialRestrictions(restriction string, event *slack.MessageEvent)
...
+		"buffalo":   {reactions: []string{"gobuffalo"}},
+		"gobuffalo": {reactions: []string{"gobuffalo"}},
+		"ghost":     {reactions: []string{"ghost"}},
+		"ermergerd": {reactions: []string{"dragon"}},
+		"ermahgerd": {reactions: []string{"dragon"}},
+		"dragon":    {reactions: []string{"dragon"}},
+		"spacex":    {reactions: []string{"rocket"}},
+		"beer me":   {reactions: []string{"beer", "beers"}},
+		"spacemacs": {reactions: []string{"spacemacs"}},
+		"emacs":     {reactions: []string{"vim"}, rand: true},
```
Comment (jadr2ddude) [inline bot/bot.go]:
```suggestion
		"emacs":     {reactions: []string{"emacs"}, rand: true},
```

File: bot/bot.go [hunk 2]
```diff
@@ -212,19 +213,25 @@ func (b *Bot) specialRestrictions(restriction string, event *slack.MessageEvent)
...
+		"vim":       {reactions: []string{"emacs"}, rand: true},
```
Comment (jadr2ddude) [inline bot/bot.go]:
```suggestion
		"vim":       {reactions: []string{"vim"}, rand: true},
```

File: bot/bot.go (hunk 1 above)
Comment (theckman) [reply bot/bot.go]: Nope.

File: bot/bot.go (hunk 2 above)
Comment (theckman) [reply bot/bot.go]: Also nope.

**Pull request merged**